HW_35/car.py

```diff
@@ -0,0 +1,79 @@
+class Car:
+    total_cars = 0
+
+    def __init__(self, make, model, fuel_capacity):
+        """Initialize a new Car instance.
+            Args:
+            make (str): The manufacturer of the car.
+            model (str): The model of the car.
+            fuel_capacity (float): The fuel capacity of the car in liters."""
+
+        self.make = make
+        self.model = model
+        self.fuel_capacity = fuel_capacity
+        self.fuel_level = 0
+        Car.total_cars += 1
+
+    def drive(self, distance):
+        """Simulate driving the car a certain distance, and reduces the fuel level based on car fuel efficiency.
+        Args:
+        distance (float): The distance to drive in kilometers."""
+
+        fuel_needed = distance / 15
+        if fuel_needed <= self.fuel_level:
+            self.fuel_level -= fuel_needed
+            print(f'Drove {distance} km. Remaining fuel: {self.fuel_level:.2f} liters.')
+        else:
+            print('Not enough fuel to drive the requested distance.')
+
+    def refuel(self, amount):
+        """Refuel the car by a certain amount of liters. Ensures that the fuel level doesn't exceed the fuel capacity.
+        Args:
+        amount (float): The amount of fuel to be added in litres"""
+
+        if Car.is_valid_fuel_amount(amount):
+            if amount + self.fuel_level <= self.fuel_capacity:
+                self.fuel_level += amount
```

Comment:
If it is possible to refill, new amount will be calculated twice. Could be calculated once:
```suggestion
            refilled_amount = self.fuel_level + amount
            if refilled_amount <= self.fuel_capacity:
                self.fuel_level = refilled_amount
```